### PR TITLE
Feature/forum open close thread implementation

### LINKED ISSUE #1488: Forum: open/close thread implementation
## Overview

When pressing the "open thread" button created in the PR solving #1482, we should open the tread.

## Acceptance criteria
- [ ] add a new `allUsersGroupId` number configuration parameter with as description `groupId of the all-users group used by the backend (used while the backend cannot provide us with it)`
- [ ] when pressing "open thread" (and the button is enabled!), call the `updateThread` service with as `status`: `waiting_for_trainer`, `helper_group_id`: the `allUsersGroupId` config value, 
- [ ] when pressing "close thread" (and the button is enabled!), call the `updateThread` service with as `status`: `closed`

On both actions: 
  - [ ] on success, force the refresh of the thread info (which should make the "open thread" button disappear)
  - [ ] on failure 4xx, display a red toast "Unable to open the thread"
  - [ ] on failure 5xx, display a red toast with the usual message

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1488

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [ ] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/forum-open-close-thread-implementation/en/a/766891051147204195;p=7523720120450464843;a=0)
  3. And I click on thread button
  4. Then I see the thread is opened
  5. And I click on "Open this thread"
  6. Then I see the thread is opened and status changed
  7. And I click on "Close this thread"
  8. Then I see the thread is close and status changed

- [ ] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/forum-open-close-thread-implementation/en/a/766891051147204195;p=7523720120450464843;a=0)
  3. And I click on thread button
  4. Then I see the thread is opened
  5. Then I turn off network connection
  6. And I click on "Open this thread" or "Close this thread"
  7. Then I see the notification error

- [ ] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/forum-open-close-thread-implementation/en/a/766891051147204195;p=7523720120450464843;a=0)
  3. And I click on thread button
  4. Then I see the thread is opened
  5. And I type new message
  6. Then I press on enter key
  7. Then I see the message is sent and status is not changed

### CODE REVIEW

File: src/app/core/components/thread/thread.component.ts [hunk 1]
```diff
@@ -165,7 +170,19 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
...
+    this.updateThreadService.update(threadInfo.itemId, threadInfo.participantId, open ? {
+      status: 'waiting_for_trainer',
+      helperGroupId: environment.allUsersGroupId,
+    } : { status: 'closed' }).subscribe({
+      next: () => this.threadService.refresh(),
+      error: error => {
+        if (errorIsHTTPForbidden(error)) {
+          this.actionFeedbackService.error($localize`Unable to open the thread`);
+          return;
+        }
```
Comment (smadbe) [inline src/app/core/components/thread/thread.component.ts:183]: Actually, as we check all possible error and disable button if the action is not possible, this block is not useful. All errors are "unexpectedError" errors.

File: src/app/core/components/thread/thread.component.html [hunk 2]
```diff
@@ -61,18 +61,21 @@
...
-              </ng-template>
-              </button>
+              <ng-container *ngIf="threadInfo$ | async as threadInfo">
+                <button
+                  type="button"
+                  class="alg-button primary small"
+                  [disabled]="!canSwitch"
+                  pButton
+                  (click)="changeThreadStatus(!threadStatus.data.open, threadInfo.data)"
+                  *ngIf="threadStatus && threadStatus.data && threadInfo.data"
```
Comment (smadbe) [inline src/app/core/components/thread/thread.component.html:71]: From what I see, you do not user threadInfo$ but just the threadId$... if so, just use `this.threadService.threadId$,` directly, that's easier (not a fetch state).

**Review by smadbe — CHANGES_REQUESTED** [review]
Good 👍

File: src/app/core/components/thread/thread.component.html (hunk 2 above)
Comment (Iloveall) [reply src/app/core/components/thread/thread.component.html]: Fixed

File: src/app/core/components/thread/thread.component.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/core/components/thread/thread.component.ts]: Fixed

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**